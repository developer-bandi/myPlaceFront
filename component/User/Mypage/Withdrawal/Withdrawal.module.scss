.subTitle {
  font-size: 25px;
  margin-bottom: 10px;
}

.content {
  margin-bottom: 30px;
  color: gray;
}

.buttonBlock {
  display: flex;
  margin-top: 30px;
}

.button {
  width: 480px;
  height: 50px;
  border-radius: 5px;
  cursor: pointer;
  @media screen and (max-width: 1200px) {
    width: calc((100vw - 80px) / 2);
  }
  &.agree {
    border: 1px solid #dddfe4;
    margin-right: 40px;
  }
  &.disagree {
    background-color: #dddfe4;
    color: white;
  }
}
