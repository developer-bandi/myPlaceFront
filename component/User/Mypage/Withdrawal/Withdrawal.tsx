import MyPageNavigation from "../Common/navigation/MyPageNavigation";
import styles from "./Withdrawal.module.scss";
import mypage from "../../../../lib/styles/mypage.module.scss";
interface WithdrawalProps {
  withdrawalButton: () => void;
  disagreeButton: () => void;
  isLabtopOrTabletOrMobile: boolean;
}

const Withdrawal = ({
  withdrawalButton,
  disagreeButton,
  isLabtopOrTabletOrMobile,
}: WithdrawalProps) => {
  return (
    <div className={mypage.mainBlock}>
      {isLabtopOrTabletOrMobile ? null : <MyPageNavigation />}
      <div className={mypage.subBlock}>
        <h1 className={mypage.title}>회원탈퇴</h1>
        <div>
          <div className={styles.contentBlock}>
            <div className={styles.subTitle}>개인정보 삭제</div>
            <div className={styles.content}>
              사용자의 아이디 닉네임과 작성한 후기 포스트 댓글이 모두
              삭제됩니다.
            </div>
          </div>
          <div className={styles.contentBlock}>
            <div className={styles.subTitle}>소셜 계정 연결 정보 삭제</div>
            <div className={styles.content}>
              소셜 계정을 연결한 경우 탈퇴 시 연결 정보도 함께 삭제됩니다.
            </div>
          </div>
        </div>
        <div className={styles.buttonBlock}>
          <button
            className={`${styles.button} ${styles.agree}`}
            onClick={withdrawalButton}
          >
            동의
          </button>
          <button
            className={`${styles.button} ${styles.disagree}`}
            onClick={disagreeButton}
          >
            비동의
          </button>
        </div>
      </div>
    </div>
  );
};

export default Withdrawal;
