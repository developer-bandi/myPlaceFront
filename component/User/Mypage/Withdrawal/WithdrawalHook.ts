import {useRouter} from "next/router";
import {axiosDeleteUser} from "../../../../lib/commonFn/api";

const useWithdrawal = () => {
  const router = useRouter();
  const withdrawalButton = async () => {
    if (window.confirm("탈퇴하시겠습니까?")) {
      try {
        const res = await axiosDeleteUser();
        if (res.status === 200) {
          alert("회원탈퇴되었습니다");
          router.push("/");
        }
      } catch (error) {
        alert("에러가 발생하였습니다");
      }
    }
  };

  const disagreeButton = () => {
    router.push("/");
  };

  return {
    withdrawalButton,
    disagreeButton,
  };
};

export default useWithdrawal;
