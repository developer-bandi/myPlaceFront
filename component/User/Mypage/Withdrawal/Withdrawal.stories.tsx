import { action } from "@storybook/addon-actions";
import { ComponentStory, ComponentMeta } from "@storybook/react";
import { useIsLabtopOrTabletOrMobile } from "../../../../lib/customHook/mediaQuery";
import Withdrawal from "./Withdrawal";

export default {
  title: "User/Mypage/Withdrawal",
  component: Withdrawal,
} as ComponentMeta<typeof Withdrawal>;

const Template: ComponentStory<typeof Withdrawal> = (arg) => {
  const isLabtopOrTabletOrMobile = useIsLabtopOrTabletOrMobile();

  return (
    <Withdrawal {...arg} isLabtopOrTabletOrMobile={isLabtopOrTabletOrMobile} />
  );
};

export const generalType = Template.bind({});
generalType.args = {
  disagreeButton: action("disagreeButton"),
  withdrawalButton: action("withdrawalButton"),
};
