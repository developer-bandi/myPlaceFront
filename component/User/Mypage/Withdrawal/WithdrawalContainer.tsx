import { useIsLabtopOrTabletOrMobile } from "../../../../hooks/mediaQuery";
import Withdrawal from "./Withdrawal";
import useWithdrawal from "./WithdrawalHook";

const WithdrawalContainer = () => {
  const isLabtopOrTabletOrMobile = useIsLabtopOrTabletOrMobile();
  const { withdrawalButton, disagreeButton } = useWithdrawal();
  return (
    <Withdrawal
      withdrawalButton={withdrawalButton}
      disagreeButton={disagreeButton}
      isLabtopOrTabletOrMobile={isLabtopOrTabletOrMobile}
    />
  );
};

export default WithdrawalContainer;
