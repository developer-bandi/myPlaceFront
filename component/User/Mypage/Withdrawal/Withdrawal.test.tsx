import {
  act,
  fireEvent,
  render,
  renderHook,
  screen,
} from "@testing-library/react";
import axios from "axios";
import { useRouter } from "next/router";
import Withdrawal from "./Withdrawal";
import useWithdrawal from "./WithdrawalHook";

jest.mock("axios");
jest.mock("next/router", () => ({
  __esModule: true,
  useRouter: jest.fn(),
}));
const mockRouter = {
  pathname: "",
  push: jest.fn(),
};
(useRouter as jest.Mock).mockReturnValue(mockRouter);
const mockedAxios = axios as jest.Mocked<typeof axios>;

describe("Withdrawal Hook 테스트", () => {
  describe("withdrawalButton 함수 테스트", () => {
    it("정상적으로 삭제된경우", async () => {
      mockedAxios.delete.mockImplementation(() =>
        Promise.resolve({
          status: 200,
          data: "data",
        })
      );
      window.confirm = jest.fn(() => true);
      window.alert = jest.fn();
      const { result } = renderHook(() => useWithdrawal());
      await act(async () => {
        await result.current.withdrawalButton();
      });
      expect(window.alert).toBeCalledWith("회원탈퇴되었습니다");
      expect(mockRouter.push).toBeCalledWith("/");
    });

    it("에러가 발생한 경우", async () => {
      mockedAxios.delete.mockImplementation(() =>
        Promise.reject({
          status: 500,
          data: "error",
        })
      );
      window.confirm = jest.fn(() => true);
      window.alert = jest.fn();
      const { result } = renderHook(() => useWithdrawal());
      await act(async () => {
        await result.current.withdrawalButton();
      });
      expect(window.alert).toBeCalledWith("에러가 발생하였습니다");
    });
  });

  it("disagreeButton 함수 테스트", () => {
    window.alert = jest.fn();
    const { result } = renderHook(() => useWithdrawal());
    act(() => {
      result.current.disagreeButton();
    });
    expect(mockRouter.push).toBeCalledWith("/");
  });
});
