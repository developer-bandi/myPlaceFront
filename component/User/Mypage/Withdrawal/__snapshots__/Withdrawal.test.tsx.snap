// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyPassword Presentational 테스트 노트북 이하 사이즈 1`] = `
<div>
  <div
    class="mainBlock"
  >
    <div
      class="mainBlock"
    >
      <div
        class="subTitle"
      >
        장소찾기
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/bookmark"
          >
            북마크
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/review"
          >
            작성 후기
          </a>
        </li>
      </ul>
      <div
        class="subTitle"
      >
        커뮤니티
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/post"
          >
            작성 글
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/comment"
          >
            작성 댓글
          </a>
        </li>
      </ul>
      <div
        class="subTitle"
      >
        계정
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/infosetting"
          >
            내 정보
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/withdrawal"
          >
            회원 탈퇴
          </a>
        </li>
      </ul>
    </div>
    <div
      class="subBlock"
    >
      <h1
        class="title"
      >
        회원탈퇴
      </h1>
      <div>
        <div
          class="contentBlock"
        >
          <div
            class="subTitle"
          >
            개인정보 삭제
          </div>
          <div
            class="content"
          >
            사용자의 아이디 닉네임과 작성한 후기 포스트 댓글이 모두 삭제됩니다.
          </div>
        </div>
        <div
          class="contentBlock"
        >
          <div
            class="subTitle"
          >
            소셜 계정 연결 정보 삭제
          </div>
          <div
            class="content"
          >
            소셜 계정을 연결한 경우 탈퇴 시 연결 정보도 함께 삭제됩니다.
          </div>
        </div>
      </div>
      <div
        class="buttonBlock"
      >
        <button
          class="button agree"
          data-testid="withdrawalButton"
        >
          동의
        </button>
        <button
          class="button disagree"
          data-testid="disagreeButton"
        >
          비동의
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`MyPassword Presentational 테스트 데스크톱 1`] = `
<div>
  <div
    class="mainBlock"
  >
    <div
      class="mainBlock"
    >
      <div
        class="subTitle"
      >
        장소찾기
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/bookmark"
          >
            북마크
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/review"
          >
            작성 후기
          </a>
        </li>
      </ul>
      <div
        class="subTitle"
      >
        커뮤니티
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/post"
          >
            작성 글
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/comment"
          >
            작성 댓글
          </a>
        </li>
      </ul>
      <div
        class="subTitle"
      >
        계정
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/infosetting"
          >
            내 정보
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/withdrawal"
          >
            회원 탈퇴
          </a>
        </li>
      </ul>
    </div>
    <div
      class="subBlock"
    >
      <h1
        class="title"
      >
        회원탈퇴
      </h1>
      <div>
        <div
          class="contentBlock"
        >
          <div
            class="subTitle"
          >
            개인정보 삭제
          </div>
          <div
            class="content"
          >
            사용자의 아이디 닉네임과 작성한 후기 포스트 댓글이 모두 삭제됩니다.
          </div>
        </div>
        <div
          class="contentBlock"
        >
          <div
            class="subTitle"
          >
            소셜 계정 연결 정보 삭제
          </div>
          <div
            class="content"
          >
            소셜 계정을 연결한 경우 탈퇴 시 연결 정보도 함께 삭제됩니다.
          </div>
        </div>
      </div>
      <div
        class="buttonBlock"
      >
        <button
          class="button agree"
          data-testid="withdrawalButton"
        >
          동의
        </button>
        <button
          class="button disagree"
          data-testid="disagreeButton"
        >
          비동의
        </button>
      </div>
    </div>
  </div>
</div>
`;
